### PR TITLE
resources: add packer files to make x86 and arm npb image

### PR DESCRIPTION
**Pull request by Harshil2107**
- This PR renames the `npb` directory to `npb-24.04-imgs`
- Adds packer files to make the x86 and arm npb images
- Adds other necessary files like `post-install.sh` that are required by the image to build the NPB benchmarks

### CODE REVIEW

File: src/x86-ubuntu/scripts/post-installation.sh
Comment (erin-le) [inline src/x86-ubuntu/scripts/post-installation.sh:1]: Is this file needed?

Comment (Harshil2107) [reply src/x86-ubuntu/scripts/post-installation.sh:1]: reverted the changes made to the file

File: src/npb-24.04-imgs/build-arm.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+PACKER_VERSION="1.10.0"
+
+if [ ! -f ./packer ]; then
+    wget https://releases.hashicorp.com/packer/${PACKER_VERSION}/packer_${PACKER_VERSION}_linux_arm64.zip;
+    unzip packer_${PACKER_VERSION}_linux_arm64.zip;
+    rm packer_${PACKER_VERSION}_linux_arm64.zip;
+fi
+
+# make the flash0.sh file
+cd ./files
```
Comment (erin-le) [inline src/npb-24.04-imgs/build-arm.sh:11]:
```suggestion
mkdir files
cd ./files
```

File: src/npb-24.04-imgs/arm-npb.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  memory           = "8192"
+  output_directory = "disk-image-arm-npb"
+  qemu_binary      = "/usr/bin/qemu-system-aarch64"
+  qemuargs         = [  ["-boot", "order=dc"],
+                        ["-bios", "./files/flash0.img"],
+                        ["-cpu", "host"],
+                        ["-enable-kvm"],
+                        ["-machine", "virt"],
+                        ["-machine", "gic-version=3"],
+                        ["-vga", "virtio"],
```
Comment (erin-le) [inline src/npb-24.04-imgs/arm-npb.pkr.hcl]:
```suggestion
```

File: src/npb-24.04-imgs/README.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+- The `gem5-bridge`(m5) utility is installed in `/usr/local/bin/gem5-bridge`.
+- `libm5` is installed in `/usr/local/lib/`.
+- The headers for `libm5` are installed in `/usr/local/include/gem5-bridge`.
+- `npb` benchmark sutie with ROI annotations
+
+Thus, you should be able to build packages on the disk and easily link to the gem5-bridge library.
+
+The disk has network disabled by default to improve boot time in gem5.
+
+If you want to enable networking, you need to modify the disk image and move the file `/etc/netplan/50-cloud-init.yaml.bak` to `/etc/netplan/50-cloud-init.yaml.bak`.
```
Comment (erin-le) [inline src/npb-24.04-imgs/README.md]: These appear to be the same file path.

File: src/npb-24.04-imgs/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+to build the x86 disk image or 
+
+```sh
+./build-arm.sh
+```
+
+to run the arm disk image.
+After this process succeeds, the disk image can be found on the `npb-24.04-imgs/disk-image-x86-npb/disk-image-x86-npb` or `npb-24.04-imgs/disk-image-arm-npb/disk-image-arm-npb` repectively.
+
+This gapbs image uses the prebuilt ubuntu 24.04 image as a base image. The gapbs image also throws the same exit events as the base image.
```
Comment (erin-le) [inline src/npb-24.04-imgs/README.md]:
```suggestion
This npb image uses the prebuilt ubuntu 24.04 image as a base image. The npb image also throws the same exit events as the base image.
```

**Review by erin-le — APPROVED** [review]
LGTM. I tested building both disk images and used them as resources to run UA size s. I had some trouble writing/getting the config scripts to work, so it could be a good idea to add  example scripts for these disk images in configs/example/gem5_library, then update the documentation to point to them.
